### PR TITLE
Allow device configuration with configuration file

### LINKED ISSUE #410: Allow device configuration with configuration file
Enhance the app and hardware configuration handling.

## Expected Behavior or Design
The following features should be implemented:
- Hardware configuration:
  - Option to enable / disable each device.   
    If a device is disabled then simply use the mock implementation.
  - Configure i2c bus and device address id in configuration file.
  - Configure GPIO pins in configuration file.
  - Enhance json schema file to allow configuration file validation.
- General:
  - Possibility to use a different app or hardware configuration file.   
    At the moment the configuration file names are hard coded.
  - Configuration profile support:
    - Load entirely different configurations based on a profile specified at app startup.

## Current Behavior or Design
Only a small part of the hardware devices can be configured in hardware.json. However, the devices are mostly prepared to be configured in the hardware factory. What is missing is some glue code, enhancing the configuration file and exposing missing parameters in the devices.

## Possible Solution
- Add command line options for different configuration files.
- Profile support:
  - Add optional command line parameter which loads different configuration files.   
     E.g. add profile name as configuration file suffix
  - Propagate profile name to concrete hardware factory implementation.   
     This allows to the factory to do whatever it needs to do for a specific profile.

### PR DESCRIPTION
**Pull request by zehnm**
Enhanced app and device configuration options and added optional command line options.

- Hardware configuration:
  - Option to enable / disable each device.   
    If a device is disabled then simply use the mock implementation.
  - Option to configure i2c bus and device address id.
  - Option to configure GPIO pins.
  - Enhanced hardware json schema file.
  - Added minimal hardware.json file for generic Linux systems:
    - All sensor devices are disabled, i.e. the mock implementation is used.
    - wpa_supplicant and systemd are active
    - Tested on Raspbian with a RPi3 and HDMI display
- Command line options:
  - Help & version (default handling by Qt).
  - Validate app & hardware configuration files and the terminate with exit code.
  - Configuration file override flags for config.json, hardware.json and their schema files.
  - Configuration file profile support.
    - Appends profile name to configuration file.
    - Profile `foobar` will try to load `config-foobar.json` and `hardware-foobar.json`

This closes #410

### CODE REVIEW

File: hardware-linux.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        "accessPoint": {
+            "TODO": true
+        }
+    },
+    "webserver" : {
+        "enabled": true,
+        "lighttpd" : {
+            "configFile": "/etc/lighttpd/lighttpd.conf",
+            "wifiSetupConfig": "/etc/lighttpd/lighttpd-wifisetup.conf",
+            "webConfiguratorConfig": "/etc/lighttpd/lighttpd-config.conf"
```
Comment (martonborzak) [inline hardware-linux.json:77]: These won't be necessary as we discussed to have the initial setup on the remote itself. So we will only server one website (the configurator).

File: hardware.json [hunk 2]
```diff
@@ -45,6 +114,7 @@
         }
     },
     "webserver" : {
+        "enabled": true,
         "lighttpd" : {
             "configFile": "/etc/lighttpd/lighttpd.conf",
             "wifiSetupConfig": "/etc/lighttpd/lighttpd-wifisetup.conf",
```
Comment (martonborzak) [inline hardware.json:120]: Same as above.

**Review by martonborzak — APPROVED** [review]

File: hardware.json (hunk 2 above)
Comment (zehnm) [reply hardware.json:120]: This configuration is required, otherwise the webserver cannot be started / stopped from the settings menu ;-)

Comment (martonborzak) [reply hardware.json:120]: I know, I'm talking about this: `"wifiSetupConfig": "/etc/lighttpd/lighttpd-wifisetup.conf",`. Just the line above would be enough, wouldn't it?

File: hardware-linux.json (hunk 1 above)
Comment (zehnm) [reply hardware-linux.json:77]: This needs a bigger cleanup later on, also in remote-os.
We should probably keep the AP functionality somehow as last-resort recovery. But let's discuss that outside of this PR

Comment (martonborzak) [reply hardware-linux.json:77]: Sure!

**Pull request merged**